src/osd/PrimaryLogPG.cc

```diff
@@ -5493,9 +5494,12 @@ int PrimaryLogPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
   ObjectState& obs = ctx->new_obs;
   object_info_t& oi = obs.oi;
   const hobject_t& soid = oi.soid;
-  bool skip_data_digest = osd->store->has_builtin_csum() &&
-    cct->_conf->osd_skip_data_digest;
-  const uint64_t osd_max_object_size = cct->_conf->osd_max_object_size;
+  const static md_config_cacher_t<uint64_t> osd_max_object_size(
+    *g_conf, "osd_max_object_size");
+  const static md_config_cacher_t<bool> conf_data_digest(
+    *g_conf, "osd_skip_data_digest");
```

Comment:
If these are static the lifecycle is not quite right. (For example, these will outlive the CephContext and md_config_t itself.)  Instead, let's make these members of OSDService